source/classes.tex

```diff
@@ -4765,10 +4765,10 @@
 ``\tcode{this->}'',
 is used to access a non-static data member or non-static
 member function, the reference is ill-formed if the
-left operand (considered as a pointer in the
+left operand is of type \cv{}~\tcode{D*} (or \cv{}~\tcode{D} in the
```

Comment:
In the core language section, we usual use the "quotation-mark phrases" for types introduced in [dcl.meaning], e.g. "pointer to cv D".

That said, I'm increasingly warming up to the idea to move this entire provision into [class.ref], where it actually belongs. Adding examples for each of the cases that can go wrong would help, too. While doing so, we could get rid of `this->` and similar oddities.

Given our lack of CWG expertise among the editors, I'm not sure this cleanup can be made editorially.

Reply:
Now I've changed to use "pointer to _cv_ `D`". But [class.ref] doesn't seem to exist...

Reply:
Sorry, I meant expr.ref.

Reply:
Moved.